mllib/src/main/scala/org/apache/spark/ml/feature/VectorIndexer.scala

```diff
@@ -345,12 +350,27 @@ class VectorIndexerModel private[ml] (
   /** @group setParam */
   def setOutputCol(value: String): this.type = set(outputCol, value)
 
+  /** @group setParam */
+  def setHandleInvalid(value: String): this.type = set(handleInvalid, value)
+  setDefault(handleInvalid, "error")
+
   override def transform(dataset: DataFrame): DataFrame = {
     transformSchema(dataset.schema, logging = true)
     val newField = prepOutputField(dataset.schema)
+
+    val filteredDataset = (getHandleInvalid) match {
+      case "skip" => {
+        val filterer = udf { label: Vector =>
+          categoryMaps.forall { case (index, categoryMap) => categoryMap.contains(label(index)) }
```

Comment:
Did you check the performance? This could be quite expensive. Another option is to catch errors in `transformFunc` and set output to null, and then filter out null values. It is less accurate but could be faster.